ee/tabby-webserver/src/service/mod.rs

```diff
@@ -41,10 +41,20 @@ struct ServerContext {
     code: Arc<dyn CodeSearch>,
 }
 
+fn to_cli_args(url: String, token: String) -> Arc<[String]> {
+    vec!["--token".to_string(), token, "--url".to_string(), url].into()
+}
+
 impl ServerContext {
-    pub async fn new(logger: Arc<dyn RawEventLogger>, code: Arc<dyn CodeSearch>) -> Self {
+    pub async fn new(
+        logger: Arc<dyn RawEventLogger>,
+        code: Arc<dyn CodeSearch>,
+        url: String,
+    ) -> Self {
         let db_conn = DbConn::new().await.unwrap();
-        run_cron(&db_conn, false).await;
+        let token = db_conn.read_registration_token().await;
```

Comment:
registeration token could be refreshed during server's lifecycle. We should read it when kicking off job.